src/object/cli_shard.c

```diff
@@ -2189,10 +2189,6 @@ obj_shard_query_key_cb(tse_task_t *task, void *data)
 	if (check && flags & DAOS_GET_RECX) {
 		bool get_max = (okqi->okqi_api_flags & DAOS_GET_MAX);
 
-		/** if first cb, set recx */
-		if (!first && !changed)
-			D_ASSERT(is_ec_obj);
-
```

Comment:
should the assertion at above L2186 also be removed?

Reply:
yes indeed since querying just akey with preset dkey would trigger the assertion. ill update.